flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/schema/DeferredTypeFlinkTableFunction.scala

```diff
@@ -40,6 +43,10 @@ class DeferredTypeFlinkTableFunction(
     val implicitResultType: TypeInformation[_])
   extends FlinkTableFunction(tableFunction, evalMethod) {
 
+  val paramTypeInfos = evalMethod.getParameterTypes.map { paramType =>
+    TypeExtractor.getForClass(paramType)
```

Comment:
The type extraction does not always work (esp. not for Row). That's why I think that passing `Class[_]` might be better.